orchestration/dagster_orchestration/hca_orchestration/pipelines/cut_snapshot.py

```diff
@@ -0,0 +1,39 @@
+from dagster import ModeDefinition, pipeline
+
+from hca_orchestration.solids.create_snapshot import create_snapshot, make_snapshot_public
+from hca_orchestration.resources import jade_data_repo_client, noop_data_repo_client
+from hca_orchestration.resources.sam import prod_sam_client, noop_sam_client
+
+
+prod_mode = ModeDefinition(
+    name="prod",
+    resource_defs={
+        "data_repo_client": jade_data_repo_client,
+        "sam_client": prod_sam_client,
+    }
+)
+
+local_mode = ModeDefinition(
+    name="local",
+    resource_defs={
+        "data_repo_client": jade_data_repo_client,
+        # we don't want to actually hit sam and make a snapshot public
+        # unless we're running in prod
+        "sam_client": noop_sam_client,
+    }
+)
+
+test_mode = ModeDefinition(
+    name="test",
+    resource_defs={
+        "data_repo_client": noop_data_repo_client,
+        "sam_client": noop_sam_client,
+    }
+)
+
+
+@pipeline(
+    mode_defs=[prod_mode, local_mode, test_mode]
+)
+def cut_snapshot() -> None:
+    make_snapshot_public(create_snapshot())
```

Comment:
I think there should be an inspection of the result of create snapshot before moving on to making it public. In the not unlikely event the snapshot fails, the pipeline should bail out and ping slack (ideally with the job error msg).

Feel free to push back on the scope creep: in reality, 100% of all snapshot failures result in a "just try again" from Jade, so I'm wondering if it'd make sense to automate that here with a backoff + retry.

Reply:
When you say 'inspection of the result,' do you mean that the inspection should be manual? If so, we won't be able to automate that.

I'm going back and adding a check to the "poll the data repo" solid to make sure it fails the solid if the job it's polling failed, which should prevent it from moving onto publicizing a failed snapshot.

Reply:
No, it should be automated as you describe.